### ISSUE TITLE
group selection 

### ISSUE THREAD

**Issue by vinutah** [iithcandle/E2GC-release#1]
Hi,

How can we reproduce the results.

Can you please upload the group selection module.

**Comment by Nandan91** [comment]
Dear Vinu

Apologies for the late reply.

Initially, we had group convolution code and pre-trained models on the
University's server (DGX) whose disk was crashed and we all lost the data.
We are now working on this and have planned to upload the
code/pretrained-models by the end of this month. Anyways, you can reproduce
the results by following methods.

**1. Power/energy measurement: **
We had used Caffe prototxt of deep learning models (MobileNetV1 and
ResNeXt) and "nvidia-smi" command to measure the power on Nvidia GPU. These
prototxt are available on the various Github repo. To measure the power of
these models with different "g" or "G" you have to set a "group" parameter
(shown below in figure) in the caffe prototxt. Here group: 32 is actually
g=32, accordingly, you can set this value for "G" (gxG = #Ifmaps)



[image: image.png]
**2. Classification accuracy: **
We trained the models in PyTorch. You just need to define a "group"
parameter in the declaration of "conv2d" as illustrated below.
[image: image.png]
[image: image.png]

(Check the implementation details at this
<https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html>)

If you have any difficulty in the above explanation, feel free to contact
us.

Thank you for showing your interest in the E2GC paper.


Thanks and regards
Nandan Kumar Jha

**Comment by vinutah** [comment]
I think I images are missing... You might have to edit comment to re-insert them.

**Comment by Nandan91** [comment]
Hi Vinutah

Here are the previously missed images (in order)
![image](https://user-images.githubusercontent.com/30919113/92991228-0a4f5d80-f500-11ea-8f20-60b66b231b24.png)

![image](https://user-images.githubusercontent.com/30919113/92991242-29e68600-f500-11ea-8ce1-55c4538f5128.png)

![image](https://user-images.githubusercontent.com/30919113/92991251-42ef3700-f500-11ea-8b8d-0bf2a59c6204.png)
